packages/rollup-core/src/types/types.ts

```diff
@@ -21,11 +18,54 @@ export interface L2ToL1Message {
 }
 
 export interface RollupTransaction {
-  nonce: number
-  gasLimit: number
-  sender: Address
+  batchIndex: number
   target: Address
   calldata: string
+  sender?: Address
+  l1MessageSender?: Address
```

Comment:
Something I've mentioned to Ben but I'll say here as well -- technically we could use the `sender` field for this value, but we added `l1MessageSender` to be more explicit. That said, technically using `queueOrigin` + `sender` would be sufficient for any L1->L2 tx or even multi-chain txs